src/ghostfat.c

```diff
@@ -266,31 +306,49 @@ void uf2_read_block (uint32_t block_no, uint8_t *data)
     // second FAT is same as the first...
     if ( sectionRelativeSector >= BPB_SECTORS_PER_FAT ) sectionRelativeSector -= BPB_SECTORS_PER_FAT;
 
+    uint16_t* data16 = (uint16_t*) (void*) data;
+
     if ( sectionRelativeSector == 0 )
     {
       // first FAT entry must match BPB MediaDescriptor
       data[0] = BPB_MEDIA_DESCRIPTOR_BYTE;
+      data[1] = 0xff;
+
+      // cluster 1 is reserved
+      data16[1] = 0xffff;
+
       // WARNING -- code presumes only one NULL .content for .UF2 file
-      //            and all non-NULL .content fit in one sector
       //            and requires it be the last element of the array
-      uint32_t const end = (NUM_FILES * FAT_ENTRY_SIZE) + (2 * FAT_ENTRY_SIZE);
-      for ( uint32_t i = 1; i < end; ++i ) data[i] = 0xff;
+      uint32_t v = 2;
+      for (uint32_t i=0; i<NUM_FILES-1; i++)
```

Comment:
This is identical to the code for the `CURRENT.UF2` file.  Both this and the following loop are both filling in the FAT chain for a given file entry.  In all cases, the file contents are stored contiguously.

So, rather than looping until `i < NUM_FILES-1`, this would loop until `i < NUM_FILES`.

If the file size field is correct for CURRENT.UF2, then this should "just work".

Reply:
fixing in my branch of this ... will provide PR when it's ready

Reply:
I saw this in the current code, though in my testing, the 0xffff for CURRENT.UF2 (NUM_FILES-1) is overwritten anyway later on, therefore I removed it.